ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -91,7 +91,7 @@ public static SourceBuilder instance() {
     final KTable<K, GenericRow> toPotentiallyMaterialize;
 
     if (forceMaterialization) {
-      // add this identity mapValues call to prevent the source-changelog
+      // materialize to prevent the source-changelog
```

Comment:
Not sure if this is necessarily accurate now, as we're just as much materializing to access offset/partition later down the line?

Reply:
Good point. We can clarify the comment with something like "besides materializing necessary pseudocolumns, we also materialize to prevent ..."